### ISSUE TITLE
Nested Record definitions in Module do not fully resolve

### ISSUE THREAD

**Issue by kade-robertson** [sinclairzx81/typebox#1105]
It seems like any nested record definition in a Module can't resolve "correctly" to a nice TypeScript definition, you instead end up seeing a bunch of Typebox internals. Autocomplete in editors (at least VSCode) still seems to work as expected, but hovering to view type information makes it hard to parse the contents.

```ts
import { Type, Static } from '@sinclair/typebox';

const Things = Type.Module({
	ThingRecord: Type.Object({
		nested: Type.Record(
			Type.String(),
			Type.Object({
				one: Type.String(),
				two: Type.Number(),
			}),
		),
	}),
});

const ThingRecord = Things.Import('ThingRecord');

// resolves to:
// {
//     nested: TInferRecord<{
//         ThingRecord: TObject<{
//             nested: TRecord<TString, TObject<{
//                 one: TString;
//                 two: TNumber;
//             }>>;
//         }>;
//     }, TString, TObject<...>, string, {
//         ...;
//     }>;
// }
type ThingRecord = Static<typeof ThingRecord>;
```

When this same definition is moved outside of a module, it resolves correctly:
```ts
import { Type, Static } from '@sinclair/typebox';

const ThingRecord = Type.Object({
	nested: Type.Record(
		Type.String(),
		Type.Object({
			one: Type.String(),
			two: Type.Number(),
		}),
	),
})

// resolves to:
// {
//     nested: {
//         [x: string]: {
//             one: string;
//             two: number;
//         };
//     };
// }
type ThingRecord = Static<typeof ThingRecord>;
```

**Comment by sinclairzx81** [comment]
@kade-robertson Hi, thanks for reporting

Have pushed a fix for this on 0.34.10.

Will close up this issue for now, but if you run into any problems, feel free to ping on this thread.
Cheers
S
